src/codegen/mod.rs

```diff
@@ -610,6 +620,38 @@ impl CodeGenerator for Type {
             TypeKind::TemplateInstantiation(ref inst) => {
                 inst.codegen(ctx, result, item)
             }
+            TypeKind::BlockPointer(inner) => {
+                let inner_item = inner.into_resolver()
+                    .through_type_refs()
+                    .resolve(ctx);
+                let name = item.canonical_name(ctx);
+
+                let inner_rust_type = {
+                    // Its possible that we have better layout information than
+                    // the inner type does, so fall back to an opaque blob based
+                    // on our layout if converting the inner item fails.
```

Comment:
This comment doesn't seem to make sense.

Reply:
It makes me a bit uncomfortable to not have a switch to preserve the old behavior by default... But I won't block on that if you really think it's not worth it.

That being said, I think we should do it if it breaks any non-objc build :)

Reply:
I'm fine, we can add a new option like `--block-signature` instead of clang's `-fblocks` if you want, and use `*const c_void` without the option.